pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -459,11 +460,18 @@ func ServerPreferredNamespacedGroupVersionResources(discoveryClient discovery.Di
 	if err != nil {
 		return results, err
 	}
+
+	allErrs := []error{}
 	for _, apiGroup := range serverGroupList.Groups {
-		preferredVersion := apiGroup.PreferredVersion
```

Comment:
it seems worrying that this can be empty... shouldn't we fix that instead of compensating here?